changes/tasks.py

```diff
@@ -524,6 +560,17 @@ def check_momconnect_change_language(self, data_fields, change):
         else:
             return []
 
+    def check_momconnect_change_msisdn(self, data_fields, change):
+        """
+        Ensures that a new msisdn is provided, and that it is a valid msisdn.
+        """
+        if "msisdn" not in data_fields:
```

Comment:
I'd go with option there.
But I'm now not sure we should do this at all, the main reason being that if the user chooses to delete their data then we will have to go through all of their changes and delete the phone numbers (and id numbers for the other changes we need to implement). :confused: